File: backend/app/schemas/community.py
"""Community schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field, field_validator


class CommunityBase(BaseModel):
    """Base community schema."""

    platform: str = Field(description="Platform: 'vk' or 'telegram'")
    external_id: str = Field(description="Platform-specific ID")
    name: str = Field(max_length=255, description="Community display name")

    @field_validator("platform")
    @classmethod
    def validate_platform(cls, v: str) -> str:
        """Validate platform value."""
        if v not in ["vk", "telegram"]:
            raise ValueError("Platform must be 'vk' or 'telegram'")
        return v


class CommunityCreate(CommunityBase):
    """Community creation schema."""

    access_token: str | None = Field(default=None, description="VK access token")
    refresh_token: str | None = Field(default=None, description="VK refresh token")
    bot_token: str | None = Field(default=None, description="Telegram bot token")


class CommunityUpdate(BaseModel):
    """Community update schema."""

    name: str | None = Field(default=None, max_length=255)


class CommunityResponse(CommunityBase):
    """Community response schema."""

    id: UUID
    is_active: bool
    token_expires_at: datetime | None
    last_sync_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CommunityListResponse(BaseModel):
    """Community list response schema."""

    data: list[CommunityResponse]
    pagination: dict

File: frontend/src/types/community.ts
/** Community types. */

export type Platform = 'vk' | 'telegram';

export interface Community {
  id: string;
  platform: Platform;
  external_id: string;
  name: string;
  is_active: boolean;
  token_expires_at: string | null;
  last_sync_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface CommunityCreate {
  platform: Platform;
  external_id: string;
  name: string;
  access_token?: string;
  refresh_token?: string;
  bot_token?: string;
}

export interface CommunityUpdate {
  name?: string;
}
